sklearn/dummy.py

```diff
@@ -21,32 +21,52 @@
 
 
 class DummyClassifier(MultiOutputMixin, ClassifierMixin, BaseEstimator):
-    """
-    DummyClassifier is a classifier that makes predictions using simple rules.
+    """DummyClassifier makes predictions that ignore the input features.
+
+    This classifier serves as a simple baseline to compare against other more
+    complex classifiers.
+
+    The specific behavior of the baseline is selected with the `strategy`
+    parameter.
+
+    All strategies make predictions that ignore the input feature values passed
+    as the `X` argument to `fit` and `predict`. However the predictions
+    typically depend on values observed in the `y` parameter passed to `fit`.
 
-    This classifier is useful as a simple baseline to compare with other
-    (real) classifiers. Do not use it for real problems.
+    Note that the "stratified" and "uniform" strategies lead to
+    non-deterministic predictions that can be rendered deterministic by setting
+    the `random_state` parameter if needed. The other strategies are naturally
+    deterministic and, once fit, always return a the same prediction for any
+    value of `X`.
 
     Read more in the :ref:`User Guide <dummy_estimators>`.
 
     .. versionadded:: 0.13
 
     Parameters
     ----------
-    strategy : {"stratified", "most_frequent", "prior", "uniform", \
+    strategy : {"most_frequent", "prior", "stratified", "uniform", \
             "constant"}, default="prior"
         Strategy to use to generate predictions.
 
-        * "stratified": generates predictions by respecting the training
-          set's class distribution.
-        * "most_frequent": always predicts the most frequent label in the
-          training set.
-        * "prior": always predicts the class that maximizes the class prior
-          (like "most_frequent") and ``predict_proba`` returns the class prior.
-        * "uniform": generates predictions uniformly at random.
+        * "most_frequent": the `predict` method always returns the most frequent
+          class label in the observed in the `y` argument passed to `fit`. The
+          `predict_proba` method returns the matching one-encoded vector.
+        * "prior": `predict` returns the class that maximizes the empirical
+          distribution observed in the `y` argument passed to `fit`
+          (like "most_frequent"). ``predict_proba`` always returns the empirical
+          class distribution of `y` also known as the empirical class prior.
+        * "stratified": the `predict` method generates random predictions by
+          drawing class labels from the empirical class prior. The
+          `predict_proba` method samples one-hot vectors a from a multinomial
+          distribution parametrized by the empirical class prior.
+          Samples each row returned by by predict_proba is there for an
+          independent and identically distributed one-hot vector.
```

Comment:
```suggestion
          Each sampled row returned by `predict_proba` is therefor an
          independent and identically distributed one-hot vector.
```
This is true only if the empirical distribution of `y` is iid.

Reply:
Why?

Reply:
Right, I misunderstood the description. `predict` as well es `predict_proba` are both iid, yes.

Maybe we can change description a bit:
```python
        """
        * "stratified": the `predict_proba` method randomly samples one-hot
          vectors from a multinomial distribution parametrized by the
          empirical class prior probabilities.
          The `predict` method returns the class label which got probability one
          in the one-hot vector of `predict_proba`.
          Each sampled row of both methods is therefore
          independent and identically distributed.
         """
```

Reply:
ok

Reply:
done.